### PR TITLE
#195 - Stage Gate Change Request Form Fields 

### LINKED ISSUE #195: Change Requests - Stage Gate Change Request Form Fields
**Background:**
In order to allow users to submit new change requests, we need a submission form. This form needs to be able to handle all the different types of change requests. There are 3 core types of change requests: standard (scope) change requests, stage gates, and activation change requests. The stage gate change requests have a specific set of form fields which we can put together in this one component.

---

**Behavior:**
The new change request stage gate change request form fields component should allows users to input the leftover budget and confirm the work package is done. Additional specifications and behaviors are as follows. They may be complex to implement, so don't be afraid to do a little Googling, a little Youtube searching, and of course, ask for help from @kevinyu328 or @jamescd18.
- The leftover budget field should be a small input text field with a prepend for the dollar ($) sign
- The leftover budget field should only accept numerical values (numbers, two decimal places max)
- The confirm done field should have a bullet list reminder of things to check as shown in the sketch
- The confirm done field should be two radio buttons for yes and no.
- Both input groups are required fields
- Ignore the "Who is required for design review?" and associated text box.

---

**Steps:**
- [ ] Create the basic component
  - [ ] Create a basic react component folder structure in the `src/components` folder
  - [ ] Create the test file, component file, and ccs module file in the component folder
  - [ ] Add basic TS functional component boilerplate to the component `.tsx` file
  - [ ] Add basic test suite `describe()` with at least one `it()` test case to the test file
  - [ ] Add the component to the `app-authenticated` component (so you can see the component when starting the app in development)
- [ ] Add the desired functionality
  - [ ] Search through [react-bootstrap](http://react-bootstrap.github.io) to find the correct template component code to use
  - [ ] Add the input fields to the component
  - [ ] Add label text for each input field
  - [ ] Configure the logic specified in the behavior section
  - [ ] Add `useState()` local variables to store state variables that need to change
- [ ] Apply styling
  - [ ] Check that the label text is styled like label text
  - [ ] Check that the layout of the input fields roughly matches the sketch
- [ ] Create test cases
  - [ ] Test that the label text is visible
  - [ ] Test that the input fields are there
  - [ ] Test that the behavior logic is working properly
- [ ] Open a PR with a screenshot of the component to get review and feedback
  - [ ] Ensure you attach a screenshot when opening a PR
  - [ ] Ensure you follow the [instructions in the contributor guide](https://github.com/Northeastern-Electric-Racing/PM-Dashboard-v2/blob/main/docs/ContributorGuide.md#creating-a-pull-request)
  - [ ] Iterate and make changes based on feedback
  - [ ] Once approved for merging, attach a screenshot of the finished component to this issue

---

**Sketch:**
The stage gate change request form fields section is highlighted with a red box around it. Notice that there is no visual styling separation between the standard change request form fields and the rest of the form fields except for the spacing and layout. Ideally [flexbox](https://getbootstrap.com/docs/4.0/utilities/flex/) would be used to ensure responsiveness.
Screen Shot 2021-05-29 at 9 52 23 PM

### PR DESCRIPTION
**Pull request by VerySeriousHireableAccount**
closes #195

### CODE REVIEW

**Comment by jamescd18** [comment]
Picture?

File: src/components/change-requests/new-change-request/new-change-request-page/stage-gate-form-fileds/stage-gate-form-fields.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is part of NER's PM Dashboard and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import { Form, FormControl, InputGroup, Row } from 'react-bootstrap';
+import { exampleAllUsers } from '../../../../../utils/lib/dummy-data';
```
Comment (jamescd18) [inline src/components/change-requests/new-change-request/new-change-request-page/stage-gate-form-fileds/stage-gate-form-fields.tsx:7]: lint errors

Comment (jamescd18) [reply src/components/change-requests/new-change-request/new-change-request-page/stage-gate-form-fileds/stage-gate-form-fields.tsx:7]: also be sure to not import things from `src/lib`. Anything coming from `utils` should be imported with an absolute package path of `'utils'`.

**Review by jamescd18 — CHANGES_REQUESTED** [review]
a few minor adjustments, but otherwise it looks great!

**Comment by VerySeriousHireableAccount** [comment]
image

File: src/components/change-requests/new-change-request/new-change-request-page/stage-gate-form-fileds/stage-gate-form-fields.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            ))}
+          </Form.Group>
+        </div>
+      </div>
+    </Form>
+  );
+};
+
+
+
```
Comment (jamescd18) [inline src/components/change-requests/new-change-request/new-change-request-page/stage-gate-form-fileds/stage-gate-form-fields.tsx:56]:
```suggestion
```

**Review by jamescd18 — APPROVED** [review]
Awesome job!

**Review by jamescd18 — APPROVED** [review]

**Pull request merged**